File: Lecture 1/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;

    // Take input from the user and verify non negative number
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);

    int cents = round(dollars * 100);

    // printf("Coin: %f\n", dollars);
    // printf("Cents: %d\n", cents);

    int coin = 0;

    // Check with 25 cent
    coin += cents / 25;
    cents = cents % 25;

    // Check with 10 cent
    coin += cents / 10;
    cents = cents % 10;

    // Check with 5 cent
    coin += cents / 5;
    cents = cents % 5;

    // Check with 1 cent
    coin += cents / 1;
    cents = cents % 1;

    printf("%d\n", coin);



}

File: Lecture 6/pset6/cash/cash.py
from cs50 import get_float

# Ask for input
while (True):
    # Get Float
    dollars = get_float("Change owed: ")
    if (dollars >= 0):
        break

# Convert to cents from dollars
cents = int(dollars * 100)

# Initialise coin variable
coin = 0

# Get coin of valuation 25
coin += cents // 25
cents = cents % 25

# Get coin of valuation 10
coin += cents // 10
cents = cents % 10

# Get coin of valuation 5
coin += cents // 5
cents = cents % 5

# Get coin of valuation 1
coin += cents // 1
cents = cents % 1

print(coin)